### PR TITLE
Stub sys_ss for vsh

### PR DESCRIPTION
**Pull request by clienthax**

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_ss.cpp [hunk 1]
```diff
@@ -144,7 +146,24 @@ s32 sys_ss_get_open_psid(vm::ptr<CellSsOpenPSID> psid)
...
 error_code sys_ss_appliance_info_manager(u32 code, vm::ptr<u8> buffer)
 {
-	sys_ss.todo("sys_ss_appliance_info_manager(code=0x%x, buffer=*0x%x)", code, buffer);
+	sys_ss.warning("sys_ss_appliance_info_manager(code=0x%x, buffer=*0x%x)", code, buffer);
+
+	switch (code)
+	{
+	case 0x19003:
+	{
+		u8 idps[] = { 0x00, 0x00, 0x00, 0x01, 0x00, 0x89, 0x00, 0x0B, 0x14, 0x00, 0xEF, 0xDD, 0xCA, 0x25, 0x52, 0x66 };
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_ss.cpp:155]: constexpr everything.

File: rpcs3/Emu/Cell/lv2/sys_ss.cpp [hunk 2]
```diff
@@ -153,33 +172,107 @@ error_code sys_ss_get_cache_of_product_mode(vm::ptr<u8> ptr)
...
+		return CELL_OK;
+	}
+	else if (cmd == 0x3002)
+	{
+		// Get time
+		if (a2 > 1)
+			return 0x80010500;
+
+		// a3 is actual output, not 100% sure, but best guess is its tb val
+		*vm::ptr<u64>::make(a3) = get_timebased_time();
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_ss.cpp:208]: vm::writeN(addr, value) where N is the number of bits.

File: rpcs3/Emu/Cell/lv2/sys_ss.cpp [hunk 3]
```diff
@@ -153,33 +172,107 @@ error_code sys_ss_get_cache_of_product_mode(vm::ptr<u8> ptr)
...
 
+	*dev = 0x190;
+
 	return CELL_OK;
 }
 
 error_code sys_ss_update_manager(u64 pkg_id, u64 a1, u64 a2, u64 a3, u64 a4, u64 a5, u64 a6)
 {
-	sys_ss.todo("sys_ss_update_manager(pkg=0x%llx, a1=0x%x, a2=0x%x, a3=0x%x, a4=0x%x, a5=0x%x, a6=0x%x)", pkg_id, a1, a2, a3, a4, a5, a6);
+	sys_ss.todo("sys_ss_update_manager(pkg=0x%llx, a1=0x%llx, a2=0x%llx, a3=0x%llx, a4=0x%llx, a5=0x%llx, a6=0x%llx)", pkg_id, a1, a2, a3, a4, a5, a6);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_ss.cpp:241]: llx is the same as x for rpcs3 string formatter, this change is a NO-OP.

File: rpcs3/Emu/Cell/lv2/sys_ss.cpp [hunk 4]
```diff
@@ -5,6 +5,8 @@
 #include "Emu/IdManager.h"
 #include "Emu/Cell/PPUThread.h"
 
+extern u64 get_timebased_time();
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_ss.cpp:8]: Emu/Cell/timer.hpp instead.

**Review by elad335 — CHANGES_REQUESTED** [review]

**Review by elad335 — APPROVED** [review]

**Comment by Megamouse** [comment]
idk. seems like all of these parameters are assumed to be always valid.
There isn't a single check for valid pointers, which looks like a hazard to me

File: rpcs3/Emu/Cell/lv2/sys_ss.cpp [hunk 5]
```diff
@@ -144,7 +145,24 @@ s32 sys_ss_get_open_psid(vm::ptr<CellSsOpenPSID> psid)
...
 {
-	sys_ss.todo("sys_ss_appliance_info_manager(code=0x%x, buffer=*0x%x)", code, buffer);
+	sys_ss.warning("sys_ss_appliance_info_manager(code=0x%x, buffer=*0x%x)", code, buffer);
+
+	switch (code)
+	{
+	case 0x19003:
+	{
+		constexpr u8 idps[] = { 0x00, 0x00, 0x00, 0x01, 0x00, 0x89, 0x00, 0x0B, 0x14, 0x00, 0xEF, 0xDD, 0xCA, 0x25, 0x52, 0x66 };
+		std::memcpy(buffer.get_ptr(), idps, 16);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_ss.cpp:155]: buffer might be null

File: rpcs3/Emu/Cell/lv2/sys_ss.cpp [hunk 6]
```diff
@@ -153,33 +171,113 @@ error_code sys_ss_get_cache_of_product_mode(vm::ptr<u8> ptr)
...
+	}
+
+	return 0x80010500;
 }
 
 error_code sys_ss_get_cache_of_flash_ext_flag(vm::ptr<u64> flag)
 {
 	sys_ss.todo("sys_ss_get_cache_of_flash_ext_flag(flag=*0x%x)", flag);
 
+	*flag = 0xFE; // nand vs nor from lsb
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_ss.cpp:224]: flag might be null

File: rpcs3/Emu/Cell/lv2/sys_ss.cpp [hunk 7]
```diff
@@ -153,33 +171,113 @@ error_code sys_ss_get_cache_of_product_mode(vm::ptr<u8> ptr)
...
+
 	return CELL_OK;
 }
 
 error_code sys_ss_get_boot_device(vm::ptr<u64> dev)
 {
 	sys_ss.todo("sys_ss_get_boot_device(dev=*0x%x)", dev);
 
+	*dev = 0x190;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_ss.cpp:233]: dev might be null

**Pull request merged**